### PR TITLE
chore(planner): fix duplicate column name

### LINKED ISSUE #5106: bug: `select 1 * 1.0` return wrong result
### Search before asking

- [X] I had searched in the [issues](https://github.com/datafuselabs/databend/issues) and found no similar issues.

### Version

8061a4725723ab0f6d3d1a9fd1efaeb3281aa5ee

### What's Wrong?

On mysql cli:
```
mysql> SELECT 1 * 1.0;
+---------+
| (1 * 1) |
+---------+
|       0 |
+---------+
1 row in set (0.04 sec)
Read 1 rows, 1 B in 0.005 sec., 217.72 rows/sec., 217.72 B/sec.
```

Query server info log:
```
2022-04-29T10:58:43.491152Z  INFO databend_query::servers::mysql::mysql_interactive_worker: Normal query: SELECT 1 * 1.0
```

```
mysql> SELECT 1 + 1.0;
+---------+
| (1 + 1) |
+---------+
|       1 |
+---------+
```

### How to Reproduce?

run `select 1 * 1.0` on MySQL cli

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

## Changelog

 
- Bug Fix
 

## Related Issues

Fixes #5106

### CODE REVIEW

File: common/datavalues/src/data_value.rs [hunk 1]
```diff
@@ -376,7 +376,7 @@ impl fmt::Display for DataValue {
         match self {
             DataValue::Null => write!(f, "NULL"),
             DataValue::Boolean(v) => write!(f, "{}", v),
-            DataValue::Float64(v) => write!(f, "{}", v),
+            DataValue::Float64(v) => write!(f, "{:?}", v),
```
Comment (BohuTANG) [inline common/datavalues/src/data_value.rs:379]: Why the `{}` not work?

Comment (sundy-li) [reply common/datavalues/src/data_value.rs:379]: `1.0` is formated to `1`. 

So `1.0 + 1` there will be duplicate column names, but their column types are different.

Comment (BohuTANG) [reply common/datavalues/src/data_value.rs:379]: Is it related to #5106 ?

Comment (sundy-li) [reply common/datavalues/src/data_value.rs:379]: Yes, it fixed #5106 , the tests case are added.

**Review by BohuTANG — APPROVED** [review]

**Comment by BohuTANG** [comment]
Some unit tests need fixed.

**Pull request merged**